### ISSUE TITLE
TypeError: Cannot read property 'utc' of undefined

### ISSUE THREAD

**Issue by makkesk8** [urish/angular-moment#194]
> This is now fixed, It was a nw.js issue.
> See: https://github.com/urish/angular-moment/pull/196 For more details.

**Comment by urish** [comment]
Can you verify that `moment` is loaded into your page successfully? 

Please try opening the chrome devtools and typing there `moment`, check what the it says...

Thx

**Comment by makkesk8** [comment]
```
moment
\moment\moment.js:15 function utils_hooks__hooks() {
        return hookCallback.apply(null, arguments);
    }
```

First thing I did before creating this post :)

**Comment by urish** [comment]
thanks, are you using the latest version of `angular-moment`?  that is 1.0.0-beta.2

**Comment by makkesk8** [comment]
used bower to install so I guess yes?

Also checked top of the js file : /* angular-moment.js / v0.10.3 / (c) 2013, 2014, 2015 Uri Shaked / MIT Licence */

**Comment by urish** [comment]
bower install does not currently pull the beta version, please try installing the beta by writing:

`bower i -S angular-moment#1.0.0-beta.2`

thanks!

**Comment by makkesk8** [comment]
```
TypeError: undefined is not a function
    at preprocessDate (angular-moment.js:396)
    at updateMoment (angular-moment.js:251)
    at Object.fn (angular-moment.js:269)
```

Still..

**Comment by urish** [comment]
Can you put a breakpoint at angular-moment.js line 98 and see what the value of `moment` is there?

thx

**Comment by makkesk8** [comment]
```
TypeError: undefined is not a function
    at angular.module.constant.constant.constant.directive.service.preprocessDate (VM1188 angular-moment.js:396)
    at updateMoment (VM1188 angular-moment.js:251)
    at Object.fn (VM1188 angular-moment.js:269)
```

**Comment by urish** [comment]
For some reason, `moment` is not defined when `angular-moment` is being loaded. Can't really tell you why it is happening without seeing the entire codebase, but if you could create a small sample that reproduces it I'd be able to look into it

**Comment by makkesk8** [comment]
I can do that certainly, But I came to believe now it might be nw.js (formerly known as node-webkit) that is the culprit here?

**Comment by urish** [comment]
Actually yes, see #104

**Comment by makkesk8** [comment]
I actually looked over that issue before I created that, and thought I post anyways. But as you can see even using node.js's require doesn't work.

**Comment by urish** [comment]
Can you try changing line 662 to read:

`		angularMoment(require('angular'), moment);`

instead of

`		angularMoment(require('angular'), require('moment'));` 

and check if it fixes the issue ?

**Comment by makkesk8** [comment]
same error still..

**Comment by urish** [comment]
Interesting... Are you using moment 2.10.3? Someone noted on #104 that upgrading to 2.10.3 helped solve a similar problem with nw.js

**Comment by makkesk8** [comment]
I was using 2.10.6, downgraded to 2.10.3, still no dice.

**Comment by urish** [comment]
So both when using bower and npm, `moment` comes out undefined in the context of angular-moment?

**Comment by makkesk8** [comment]
That's right.

**Comment by urish** [comment]
Okay, do you want to try debugging it and coming back with some conclusions why it happens? Alternatively, if you can create a small example that reproduces this behavior, I can look into it...

**Comment by makkesk8** [comment]
Okay, kind of "solved" it... 

```
var moment1 = require('moment');
.constant('moment', moment1)
```

But got a deprecation warning saying moment falls back to js Date and everything becomes "Invalid date."

**Comment by urish** [comment]
Do you get the warning when using `$scope.date = new Date()`? 

The deprecation warnings appears when you pass `moment` a string instead of a Date object...

**Comment by makkesk8** [comment]
> Do you get the warning when using $scope.date = new Date()?

Yes.

> The deprecation warnings appears when you pass moment a string instead of a Date object...
Yea I read about that @ the linked github issue.

**Comment by urish** [comment]
Seems like others are also facing similar issues, e.g. nwjs/nw.js#2075.

Very strange that the Date stuff doesn't work - can you please double-verify by logging the value which is passed to `preprocessDate` in line 396?

**Comment by makkesk8** [comment]
It returns nothing, its blank. Thought I broke it at first but its just nothing... It no longer says invalid date though... but nothing gets returned.

**Comment by urish** [comment]
and what does it get (inside the `value` parameter, in line 396)?

does `value instanceof Date` evaluate true at that line?

Uri

**Comment by makkesk8** [comment]
I feel embarrassed... Sorry for taking up your time on this latest issue of the new Date() thing... I accidentally closed down my powershell which had gulp watch running... Took me a while to figure out why the hell my js didnt update... Well I got the am-time-ago working in a ng-repeat a little while ago and I was stunned the new Date() thing didn't... So again I'm sorry..

However, Now when its working wonders I would like to give me two thanks for helping me with this, you've been really helpful! Not many people I know are being this helpful and actually taking the time to resolve it.

I would also like to add, If you could do a check for moment already before:
```
function angularMoment(angular, moment)
```
And then throw an error if you can't find it, Would make sense right? it would be really helpful for others. And also, document the solution I had to use to make it work for nw.js would also be helpful for others.

Again, thank you very much!

**Comment by urish** [comment]
No worries :)

Yes, I think that adding a check at the beginning of the function and throwing a more descriptive error if `moment` is undefined makes a lot of sense. If you have some time to create a PR for it, that will be awesome!

Uri

**Comment by makkesk8** [comment]
PR submitted.

**Comment by urish** [comment]
great!

**Comment by urish** [comment]
Fixed by #196
